code/game/world.dm

```diff
@@ -17,7 +22,24 @@ var/global/list/map_transition_config = MAP_TRANSITION_CONFIG
 	GLOB.timezoneOffset = text2num(time2text(0, "hh")) * 36000
 
 	makeDatumRefLists()
-	callHook("startup")
+	// Begin STUFF_THAT_BELONGS_IN_A_SUBSYSTEM_BUT_FUCK IT
+	// Its here becuase this is where the callhook used to be
+	// No really this should be in shit like SSmisc but fuck it
+	// If I get told to I will, but for now this can stay as a mess -affected
+	load_mode()
```

Comment:
![image](https://user-images.githubusercontent.com/25063394/57165701-cc143b00-6def-11e9-83fa-7a47a657b3b1.png)

Reply:
I feel like it may be better to employ the COMSIG system here, which actually works a lot like hooks to. That way you don't need this ugly list of procs.

Reply:
`COSMIG` system?

Reply:
https://github.com/ParadiseSS13/Paradise/search?utf8=%E2%9C%93&q=comsig&type=

You can register a signal. Then when a certain event happens, you can choose to send a signal to all the procs that have been registered previously.

Reply:
(To me anyway) This just looks like using multiple names or different names for procs, unless you can register two signals at once. It stall falls under the issue of making separate calls for things that are already in though.

Reply:
You can register multiple signals, that will all be called by COMSIG. And yes, but I don't think that's necessarily a bad thing.

Reply:
Fox told me to kill hooks so I am doing it. Really people should just be putting shit in the right place, or make a proc that can call multiple procs. With stuff like the VSCode langserver and global searching, these things really shouldnt be existing

Reply:
Eh, it's your call. It'd need checking anyway to see if COMSIG works so early in initialization.